src/main.cpp

```diff
@@ -5513,7 +5459,7 @@ bool SendMessages(CNode* pto, bool fSendTrickle)
                 vGetData.clear();
             }
 
-            // mapAlreadyAskedFor[inv] = nNow; //TODO: check why this was here
+            mapAlreadyAskedFor[inv] = nNow;
```

Comment:
I'm not sure why this line was commented-out when adding the net handling for the scraper messages. In effect, it resets the request retry delay for an object (block, tx, scraper message, etc.) so that a node will attempt to download it again in the next cycle if something calls `AskFor()` for that object/peer again. Without this, a node might push off a failed `getdata` request forever as described earlier. 

This is probably the only change needed to fix the problem targeted by this PR, but the explicit case that I added for orphan blocks should improve a node's responsiveness to orphan blocks.

Much of the old net messaging code from Bitcoin is pretty creaky at this point, and things like the flat two-minute back-off and this line that negates it don't make much sense. This PR should get us by until we can merge the newer messaging code.

Reply:
Maybe the next major backport we do is the basic net code from Bitcoin....

Reply:
It's one of the areas that needs the most work. Unfortunately, the net code is the top of the stack in Bitcoin's design, and it depends on just about everything else in the application. From what I've played with so far, it will be very difficult to backport net code if we don't do several other areas first. We will spend a lot of time converting the newer code for the other legacy areas just to convert it back later.